setup.py

```diff
@@ -88,6 +88,7 @@ def get_git_sha() -> str:
         "humanize",
         "itsdangerous>=1.0.0, <2.0.0",  # https://github.com/apache/superset/pull/14627
         "isodate",
+        "keys-management>=0.1.5, <0.2.0",
```

Comment:
This still has a hard requirement on the `keys-management` lib. Instead of introducing this dependency, downstream projects can drag this in on their own.

Reply:
we offer to use keys-management just for the interface part of it, not the implementation side.
still, any downstream projects can implement their own version of it

Reply:
I think that rather than dragging in this dependency, we should introduce a set of hooks, similar to this PR: #16284, which can then be plugged into by folks downstream to use whatever libs they want.